src/passes/Outlining.cpp

```diff
@@ -208,6 +208,13 @@ struct ReconstructStringifyWalker
     Function* outlinedFunc =
       getModule()->getFunction(sequences[seqCounter].func);
     ASSERT_OK(outlinedBuilder.visitFunctionStart(outlinedFunc));
+    // If the last instruction of the outlined sequence is unreachable, insert
+    // an unreachable instruction immediately after the call to the outlined
+    // function. This maintains the unreachable type in the original scope
+    // of the outlined sequence.
+    if (sequences[seqCounter].endsTypeUnreachable) {
+      ASSERT_OK(existingBuilder.makeUnreachable());
+    }
```

Comment:
This will put an unreachable before the call to the outlined function, right? This doesn't seem correct because the outlined function could have some side effects before the unreachable part and it would be incorrect to trap before executing those side effects.

Reply:
you're right, it was in the wrong spot, thanks!